CRM/Core/BAO/CustomField.php

```diff
@@ -152,42 +152,10 @@ public static function dataToHtml() {
    * @return CRM_Core_DAO_CustomField
    */
   public static function create($params) {
-    $transaction = new CRM_Core_Transaction();
-    $origParams = array_merge([], $params);
-    $params = self::prepareCreate($params);
-
-    $customField = new CRM_Core_DAO_CustomField();
-    $customField->copyValues($params);
-    $customField->save();
-
-    // make sure all values are present in the object for further processing
-    $customField->find(TRUE);
-
-    $triggerRebuild = CRM_Utils_Array::value('triggerRebuild', $params, TRUE);
-    //create/drop the index when we toggle the is_searchable flag
+    $customField = self::createCustomFieldRecord($params);
     $op = empty($params['id']) ? 'add' : 'modify';
```

Comment:
Actually @pradpnayak the net effect here is to set the op AFTER the pre hook has been called. I'm now on the fence about that - I guess it's a good thing to set afterwards even though other BAO don't but are also kinda inconsistent.